docs/root/intro/arch_overview/upstream/aggregate_cluster.rst

```diff
@@ -0,0 +1,61 @@
+.. _arch_overview_aggregate_cluster:
+
+Aggregate Cluster
+=================
+
+Aggregate cluster is used for failover between clusters from different service discovery mechanism, e.g., 
+failover from EDS upstream cluster to STRICT_DNS upstream cluster. Aggregate cluster loosely couples multiple 
+clusters by referencing their name in the :ref:`configuration <envoy_api_msg_config.cluster.aggregate.ClusterConfig>`. 
+The fallback priority is defined implicitly by the ordering in the :ref:`clusters <envoy_api_field_config.cluster.aggregate.ClusterConfig.clusters>`.
+Aggregate cluster uses tiered load balancing. The load balancer chooses cluster and piority first and then delegates the load balancing from that priority
+to the load balancer of the selected cluster. The top level load balancer reuses the existing load balancing algorithm by linearizing the 
+priority set of multiple clusters into one. 
+
+Linearize Priority Set
+----------------------
+
+Upstream hosts are divided into multiple :ref:`priority levels <arch_overview_load_balancing_priority_levels>` and each priority level contains 
+a list of healthy, degraded and unhealthy hosts. Linearization is used to simplify the host selection during load balancing by merging priority levels 
+from multiple clusters. For example, primary cluster has 3 priority levels, secondary has 2 and tertiary has 2 and the failover ordering is 
```

Comment:
What is the exact linearization algorithm?

Reply:
Should we tell user the detail of the algorithm or just show them the result of the linearization algorithm?

Reply:
The user needs to know enough detail to be able to build a model in Python or something I think. The other LB docs provide both examples and an abstract description. The user should not be told we pick hosts twice though, can you provide a suitable abstract algorithm without this?

Reply:
Add a load balancing example with abstract descriptions.